src/js/components/AdminPanels.tsx

```diff
@@ -115,19 +115,44 @@ export default function AdminPanels({
 
 export function RedisPanel({
   menus,
+  setMenus,
   redisEndpointUrl,
   redisKeyPrefix,
 }: RedisPanelProps) {
-  const handleItemSelect = async (menuKey: string, item: MenuItem) => {
+  const handleItemSelect = async (menuKey: string, item: string) => {
     try {
-      await simplePost(redisEndpointUrl, { key: menuKey, value: item.value })
-      console.log("Redis updated successfully")
+      await simplePost(redisEndpointUrl, { key: menuKey, value: item })
     } catch (error) {
       console.error("Error posting to redis:", error)
       throw error
     }
   }
 
+  useEffect(() => {
+    // This effect is used to fetch selected items from the server
+    // It runs only once when the component mounts
+    async function fetchSelectedValues() {
+      try {
+        const dataStr = await simpleGet(`${redisEndpointUrl}/controlvalues`)
+        const data = JSON.parse(dataStr) as Array<{
+          key: string
+          value: string
+        }>
+        // Update menus with selected items from server response
+        const updatedMenus = menus.map((menu) => {
+          const selectedValue = data.find(
+            (item) => item.key === menu.key
+          )?.value
+          return { ...menu, selectedItem: selectedValue || undefined }
```

Comment:
why is the `|| undefined` conditional added? My initial thought is that the `data.find(...)` above would already return `undefined` if not finding anything.